src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php

```diff
@@ -46,29 +46,36 @@ public function __construct(UrlGeneratorInterface $router = null, int $httpPort
      * In case the route name is empty, the status code will be 404 when permanent is false
      * and 410 otherwise.
      *
-     * @param Request    $request          The request instance
-     * @param string     $route            The route name to redirect to
-     * @param bool       $permanent        Whether the redirection is permanent
-     * @param bool|array $ignoreAttributes Whether to ignore attributes or an array of attributes to ignore
+     * @param Request    $request           The request instance
+     * @param string     $route             The route name to redirect to
+     * @param bool       $permanent         Whether the redirection is permanent
+     * @param bool|array $ignoreAttributes  Whether to ignore attributes or an array of attributes to ignore
+     * @param bool       $keepRequestMethod Wheter redirect action should keep HTTP request method
      *
      * @throws HttpException In case the route name is empty
      */
-    public function redirectAction(Request $request, string $route, bool $permanent = false, $ignoreAttributes = false): Response
+    public function redirectAction(Request $request, string $route, bool $permanent = false, $ignoreAttributes = false, bool $keepRequestMethod = false): Response
     {
         if ('' == $route) {
-            throw new HttpException($permanent ? 410 : 404);
+            throw new HttpException($permanent ? Response::HTTP_GONE : Response::HTTP_NOT_FOUND);
```

Comment:
we prefer explicit numbers; please revert, and use numbers everywhere in the PR

Reply:
@nicolas-grekas Change reverted